comm.go

```diff
@@ -77,9 +77,18 @@ func (p *PubSub) handleNewStream(s network.Stream) {
 			return
 		}
 
+		var rpc RPC
```

Comment:
a bit concerned about this being allocated on the stack, but i dont think the go compiler is that stupid.

Reply:
Allocating this on the stack from my side was stupid... 
Making it like it was in the past `new(RPC)` to allocate on the heap directly

Reply:
It is not that stupid, actually. `go build -gcflags="-m"` showed no escape. Still, better to keep it clear via `new`